src/test/java/com/exasol/adapter/document/files/AbsDocumentFilesAdapterIT.java

```diff
@@ -0,0 +1,116 @@
+import com.exasol.exasoltestsetup.ExasolTestSetup;
+import com.exasol.exasoltestsetup.ExasolTestSetupFactory;
+import com.exasol.exasoltestsetup.testcontainers.ExasolTestcontainerTestSetup;
+
+@Tag("integration")
+@Testcontainers
+class AbsDocumentFilesAdapterIT extends AbstractDocumentFilesAdapterIT {
+    private static IntegrationTestSetup SETUP;
+    private static TestContainer testContainer;
+    private static AbsTestSetup absTestSetup;
+
+    @BeforeAll
+    static void beforeAll() throws Exception {
+
+        final ExasolTestSetup exasolTestSetup = new ExasolTestSetupFactory(
+                Path.of("cloudSetup/generated/testConfig.json")).getTestSetup();
+        //since this file doesn't exist we just get a local exasol test container setup back
+        //ABS - and this will return a 'local' azure blob storage, using the azurite emulator.
+        absTestSetup = getAbsTestSetup(exasolTestSetup);
+        //this will set up a new Azure Blob Storage test container for us
+        testContainer = new TestContainer(absTestSetup);
+        //Exasol
+        SETUP = new IntegrationTestSetup(exasolTestSetup, absTestSetup, testContainer.getBlobContainerClient());
+    }
+
+    @NotNull
+    private static AbsTestSetup getAbsTestSetup(final ExasolTestSetup exasolTestSetup) {
+        if (exasolTestSetup instanceof ExasolTestcontainerTestSetup) {
+            return new LocalAbsTestSetup();
+        } else {
+            return new OnlineAbsTestSetup();
+        }
+    }
+
+    @AfterAll
+    static void afterAll() throws Exception {
+        testContainer.close();
+        SETUP.close();
+        absTestSetup.close();
+    }
+
+    @AfterEach
+    void after() {
+        testContainer.empty();
+        SETUP.dropCreatedObjects();
+    }
+
+    @Override
+    protected Statement getStatement() {
+        return SETUP.getExasolStatement();
+    }
+
+    @Override
+    protected void uploadDataFile(final Supplier<InputStream> fileContent, final String fileName) {
+        try (final InputStream inputStream = fileContent.get()) {
+            final Path tempFile = Files.createTempFile("data-file", ".data");
+            Files.copy(inputStream, tempFile, StandardCopyOption.REPLACE_EXISTING);
+            uploadDataFile(tempFile, fileName);
+            Files.delete(tempFile);
+        } catch (final IOException exception) {
+            throw new IllegalStateException("Filed to upload test file.", exception);
+        }
+    }
+
+    @Override
+    protected void uploadDataFile(final Path fileContent, final String fileName) {
+        try (final FileInputStream fileInputStream = new FileInputStream(fileContent.toFile())) {
+            testContainer.getBlobContainerClient().getBlobClient(fileName).upload(BinaryData.fromStream(fileInputStream));
+        } catch (final IOException exception) {
+            throw new IllegalStateException("Failed to read test-file", exception);
+        }
+    }
```

Comment:
Turn it arround. Implement uplkoading int the first method and let the secons one use it.

Reply:
Not sure what you mean but I basically copied over the tests from the gcp project. I'll leave it as it is for now.